deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/FileSystemDeploymentService.java

```diff
@@ -590,6 +612,27 @@ private void executeScannerTasks(List<ScannerTask> scannerTasks, DeploymentOpera
         }
     }
 
+    private class ManualScanCallable implements Callable<List<?>> {
```

Comment:
If you change this to remove the rollBack() method from ScannerTask, might as well make this implement Callable<List<ScannerTask>>.

Actually it could just implement Runnable, as nothing is done with the return value from call(). (I think doing nothing is correct. I think all that could be done is to schedule a rescan if scanResult.scheduleRescan == true, but doing that is wrong IMHO. If the scanner isn't enabled for automatic scanning, it shouldn't do any scanning without an explicit command.)